<launch>

  <!-- Machines File. Set to the correct robot! -->
  <include file="$(find pr2_alpha)/pre.machine" />

  <!-- Stereocam -->
  <group ns="stereo">
    <param name="videre_mode" type="str" value="none"/>
    <param name="do_colorize" type="bool" value="False"/>
    <param name="do_rectify" type="bool" value="True"/>
    <param name="do_stereo" type="bool" value="False"/>
    <param name="do_calc_points" type="bool" value="False"/>
    <param name="do_keep_coords" type="bool" value="False"/>
    <param name="fps" type="double" value="4.0"/>
    <param name="frame_id" type="string" value="stereo_optical_frame"/>
    <param name="exposure_auto" type="bool" value="true"/>
    <param name="brightness_auto" type="bool" value="true"/>
    <param name="gain_auto" type="bool" value="true"/>
    <param name="num_disp" value="128"/>
  </group>
  <node machine="four" pkg="dcam" type="stereodcam" respawn="false"/>
  <node machine="four" pkg="stereo_image_proc"
        type="stereoproc" respawn="false" />

  <!-- Stereo Checkerboard Detector -->
  <node machine="four" pkg="stereo_checkerboard_detector"
        type="stereo_checkerboard_node" respawn="false" />

  <!-- *************** Laser Pipeline **************-->

  <!-- Laser Controller -->
  <node pkg="mechanism_control" type="spawner.py"
        args="$(find pr2_experimental_controllers)/laser_tilt/laser_tilt_controller.xml" respawn="false" output="screen"/>
  <node pkg="pr2_mechanism_controllers" type="send_periodic_cmd_srv.py"
        args="laser_tilt_controller linear 30 .75 .35"
        output="screen" />

  <!-- Laser Imaging -->
  <node machine="three" pkg="dense_laser_assembler"
  	type="assembler_node.py" respawn="false" />
  <node machine="three" pkg="dense_laser_assembler"
  	type="laser_image_node.py" respawn="false" />
  <node machine="three" pkg="stereo_checkerboard_detector"
  	type="checkerboard_corners_node" respawn="false">
    <remap from="image" to="dense_tilt_scan/intensity_image" />
    <remap from="corners" to="dense_tilt_scan/pixel_corners" />
  </node>

  <!-- Extract joint values from corners -->
  <node machine="three" pkg="laser_cb_processing"
   	type="laser_cb_processing" respawn="false" />

  <!-- Project corner joint info into world -->
  <node machine="three" pkg="laser_cb_processing"
  	type="pr2_tilt_laser_projector" respawn="false" />

</launch>